test/e2e/util/util.go

```diff
@@ -1506,6 +1508,282 @@ func EnsureGuestWebhooksValidated(t *testing.T, ctx context.Context, guestClient
 	})
 }
 
+func EnsureGlobalPullSecret(t *testing.T, ctx context.Context, mgmtClient crclient.Client, guestClient crclient.Client, entryHostedCluster *hyperv1.HostedCluster) error {
+	//AtLeast(t, Version420)
+
+	var (
+		dummyImageTagMultiarch = "quay.io/hypershift/sleep:multiarch"
+		dummyImageTag12        = "quay.io/hypershift/sleep:1.2.0"
+		err                    error
+
+		// Additional Pull Secret
+		registrySecretName      = "additional-pull-secret"
+		registryNamespace       = "kube-system"
+		dummyPullSecretData     = []byte(`{"auths": {"quay.io": {"auth": "YWRtaW46cGFzc3dvcmQ="}}}`)
+		readOnlyE2EPullSecret   = []byte(`{"auths": {"quay.io": {"auth": "aHlwZXJzaGlmdCtlMmVfcmVhZG9ubHk6R1U2V0ZDTzVaVkJHVDJPREE1VVAxT0lCOVlNMFg2TlY0UkZCT1lJSjE3TDBWOFpTVlFGVE5BS0daNTNNQVAzRA=="}}}`)
+		oldglobalPullSecretData []byte
+	)
+	g := NewWithT(t)
+
+	// Create the additional-pull-secret secret in the DataPlane using the dummy pull secret.
+	// The dummy pull secret is authorized to pull restricted images.
+	err = createAdditionalPullSecret(ctx, guestClient, dummyPullSecretData, registrySecretName, registryNamespace)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to create additional-pull-secret secret")
+
+	// Check if additional-pull-secret secret is in the right place at Dataplane
+	t.Run("Check if additional-pull-secret secret is in the right place at Dataplane", func(t *testing.T) {
+		t.Log("Checking if additional-pull-secret secret is in the right place at Dataplane")
+		additionalPullSecret := hccomanifests.UserProvidedPullSecret()
+		g.Eventually(func() error {
+			if err := guestClient.Get(ctx, client.ObjectKey{Name: additionalPullSecret.Name, Namespace: additionalPullSecret.Namespace}, additionalPullSecret); err != nil {
+				return err
+			}
+			return nil
+		}, 20*time.Minute, 5*time.Second).Should(Succeed(), "additional-pull-secret secret is not present")
+		g.Expect(additionalPullSecret.Data).NotTo(BeEmpty(), "additional-pull-secret secret is empty")
+		g.Expect(additionalPullSecret.Data[corev1.DockerConfigJsonKey]).NotTo(BeEmpty(), "additional-pull-secret secret is empty")
```

Comment:
you can also use e2eutil.EventuallyObject for this secret and predicate which has some guardrails and logging for failures